LandSandBoat/server
# HP & MP calculation audit

**_I affirm:_**

## What does this pull request do?

Audit HP/MP calculations (incl curse, weakness, hp%, food)

Add new mods for weakness/curse as they are not additive to hpp/mpp
Move everything to correct places based on retail testing
Move HP/MP traits to "base" HP/MP via new mods
Add Max HP Boost II

Closes #6453
Closes #5127

## Steps to test these changes

Use weakness, curse, food, hp mods, hp% mods, merits and verify the following:

floor(floor(floor((base HP/MP ) * weakness modifier + flat HP/MP) * curse modifier + Convert HP/MP) * (1 + HPP/MPP / 100)) 
take that result, modify it by food HPP/MPP (and bounds check it) then add that to the result

Base HP/MP includes merits and job traits
